Submitted By: Tushar Teredesai <tushar@linuxfromscratch.org>
Date: 2003-10-27
Initial Package Version: 1.1.0
Origin: holodeck (yep, that's my workstataion hostname)
Description: Allows use of System installed versions for some packages instead of using
the ones included in OpenOffice. Saves space and time :-) Apply the patch and set the
proper env vars to select the system version (it is best to add the vars at the end of
Env file generated by configure). The version in the bracket indicates the version against
which the patch was tested.
* Berkeley DB (Version 4.1.25)
    export SYSTEM_DB=YES
    export SYSTEM_DB_VERSION=4.1
    export SYSTEM_DB_CLASSPATH=/usr/lib/db.jar
* cURL (Version 7.10.6)	
    export SYSTEM_CURL=YES
* expat (Version 1.95.6)
    export SYSTEM_EXPAT=YES
* freetype (Version 2.1.4)
    export SYSTEM_FREETYPE=YES
    export SYSTEM_FREETYPE_INCLUDES=`freetype-config --cflags`
* jpeg (Version 6b) -> Not yet stable
    export SYSTEM_JPEG=YES
* nas (Version 1.6)
	export SYSTEM_NAS=YES
* python (Version 2.3.2) -> Not complete.
    export SYSTEM_PYTHON=YES
    export SYSTEM_PYTHON_INCLUDES=-I/usr/include/python2.3
* zlib (Version 1.1.4)
    export SYSTEM_ZLIB=YES

WARNING: Note that this patch is not yet complete. It is also not completely
tested. The patch is a work in progress. Use at your own risk.

The patch was tested with all packages installed with a prefix of /usr and without
munging any locations where the files are installed. If the header files or libs
are in a location different from the usual directories where gcc or ld search for
them, add the proper flags to the generated Env file before starting compilation.

--- oo_1.1_src.orig/berkeleydb/makefile.mk	2003-06-12 04:50:38.000000000 -0500
+++ oo_1.1_src/berkeleydb/makefile.mk	2003-10-25 20:05:43.000000000 -0500
@@ -65,6 +65,17 @@
 PRJNAME=so_berkeleydb
 TARGET=so_berkeleydb
 
+.IF "$(SYSTEM_DB)" == "YES"
+all:
+	@echo "Using System Installed Berkeley DB..."
+	+$(COPY) $(SYSTEM_DB_CLASSPATH) $(BIN)
+.IF "$(SYSTEM_DB_VERSION)" == "4.1"
+	+$(COPY) ../xmlhelp/source/com/sun/star/help/CreateDb-4.1.java ../xmlhelp/source/com/sun/star/help/CreateDb.java
+	+$(COPY) ../xmlhelp/source/com/sun/star/help/HelpDatabases-4.1.java ../xmlhelp/source/com/sun/star/help/HelpDatabases.java
+	+$(COPY) ../xmlhelp/source/com/sun/star/help/HelpIndexer-4.1.java ../xmlhelp/source/com/sun/star/help/HelpIndexer.java
+.ENDIF
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/connectivity/prj/build.lst oo_1.1_src/connectivity/prj/build.lst
--- oo_1.1_src.orig/connectivity/prj/build.lst	2003-10-02 21:50:58.000000000 -0500
+++ oo_1.1_src/connectivity/prj/build.lst	2003-10-24 23:59:38.000000000 -0500
@@ -1,4 +1,4 @@
-cn	connectivity	:	comphelper moz svtools unixODBC NULL
+cn	connectivity	:	comphelper svtools unixODBC NULL
 cn	connectivity							usr1	-	all	cn_mkout NULL
 cn	connectivity\source\types				nmake	-	all	cn_types NULL
 cn	connectivity\source\commontools			nmake	-	all	cn_cmtools cn_parse cn_types NULL
diff -ur oo_1.1_src.orig/curl/makefile.mk oo_1.1_src/curl/makefile.mk
--- oo_1.1_src.orig/curl/makefile.mk	2003-07-16 12:24:06.000000000 -0500
+++ oo_1.1_src/curl/makefile.mk	2003-10-24 21:21:17.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=so_curl
 TARGET=so_curl
 
+.IF "$(SYSTEM_CURL)" == "YES"
+all:
+	@echo "Using System Installed cURL..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/expat/makefile.mk oo_1.1_src/expat/makefile.mk
--- oo_1.1_src.orig/expat/makefile.mk	2002-07-09 07:24:30.000000000 -0500
+++ oo_1.1_src/expat/makefile.mk	2003-10-24 21:23:17.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=so_expat
 TARGET=so_expat
 
+.IF "$(SYSTEM_EXPAT)" == "YES"
+all:
+	@echo "Using System Installed expat..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/external/gcc3_specific/makefile.mk oo_1.1_src/external/gcc3_specific/makefile.mk
--- oo_1.1_src.orig/external/gcc3_specific/makefile.mk	2003-06-18 07:44:10.000000000 -0500
+++ oo_1.1_src/external/gcc3_specific/makefile.mk	2003-10-24 21:53:33.000000000 -0500
@@ -12,6 +12,7 @@
 .IF "$(OS)"!="MACOSX"
 .IF "$(OS)"!="IRIX"
 .IF "$(OS)"!="FREEBSD"
+.IF "$(OS)"!="LINUX"
 
 
 all : $(SOLARLIBDIR)$/libstdc++.so.$(SHORTSTDCPP3) $(SOLARLIBDIR)$/libgcc_s.so.1 $(SOLARLIBDIR)$/libgcc_s.so 
@@ -28,6 +29,7 @@
         +ln -s libgcc_s.so.1 $@
 
 .ENDIF
+.ENDIF
 .ENDIF 
 .ENDIF 
 
diff -ur oo_1.1_src.orig/external/glibc/makefile.mk oo_1.1_src/external/glibc/makefile.mk
--- oo_1.1_src.orig/external/glibc/makefile.mk	2003-06-18 07:44:11.000000000 -0500
+++ oo_1.1_src/external/glibc/makefile.mk	2003-10-24 21:50:29.000000000 -0500
@@ -60,6 +60,11 @@
 #
 #*************************************************************************
 
+.IF "$(OS)" == "LINUX"
+all:
+	@echo "Using System Installed glibc..."
+.ENDIF
+
 MAKEFILERC=yes
 
 .IF "$(USE_SHELL)" == "4nt"
diff -ur oo_1.1_src.orig/external/glibc/rtufiles/makefile.mk oo_1.1_src/external/glibc/rtufiles/makefile.mk
--- oo_1.1_src.orig/external/glibc/rtufiles/makefile.mk	2003-10-02 22:19:24.000000000 -0500
+++ oo_1.1_src/external/glibc/rtufiles/makefile.mk	2003-10-24 21:50:35.000000000 -0500
@@ -60,6 +60,13 @@
 #
 #*************************************************************************
 
+.IF "$(OS)" == "LINUX"
+all:
+	@echo "Using System Installed glibc..."
+	-$(RM) config.h
+	-$(RM) getopt.h
+.ENDIF
+
 PRJ=..$/..
 
 PRJNAME=3rdparty
diff -ur oo_1.1_src.orig/external/glibc/sdfiles/makefile.mk oo_1.1_src/external/glibc/sdfiles/makefile.mk
--- oo_1.1_src.orig/external/glibc/sdfiles/makefile.mk	2001-04-09 10:20:01.000000000 -0500
+++ oo_1.1_src/external/glibc/sdfiles/makefile.mk	2003-10-24 21:32:47.000000000 -0500
@@ -60,6 +60,11 @@
 #
 #*************************************************************************
 
+.IF "$(OS)" == "LINUX"
+all:
+	@echo "Using System Installed glibc..."
+.ENDIF
+
 PRJ=..$/..
 
 PRJNAME=3rdparty
diff -ur oo_1.1_src.orig/filter/source/flash/makefile.mk oo_1.1_src/filter/source/flash/makefile.mk
--- oo_1.1_src.orig/filter/source/flash/makefile.mk	2003-07-09 05:14:43.000000000 -0500
+++ oo_1.1_src/filter/source/flash/makefile.mk	2003-10-26 08:24:26.000000000 -0600
@@ -71,6 +71,10 @@
 
 .INCLUDE :  	settings.mk
 
+.IF "$(SYSTEM_ZLIB)" == "YES"
+CFLAGS += -DSYSTEM_ZLIB
+.ENDIF
+
 # --- Types -------------------------------------
 
 UNOTYPES=\
diff -ur oo_1.1_src.orig/filter/source/flash/swfwriter1.cxx oo_1.1_src/filter/source/flash/swfwriter1.cxx
--- oo_1.1_src.orig/filter/source/flash/swfwriter1.cxx	2003-07-11 05:22:41.000000000 -0500
+++ oo_1.1_src/filter/source/flash/swfwriter1.cxx	2003-10-26 08:25:47.000000000 -0600
@@ -88,7 +88,11 @@
 #include <svtools/graphictools.hxx>
 #endif
 
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#else
 #include <external/zlib/zlib.h>
+#endif
 
 #ifndef _SV_SALBTYPE_HXX
 #include <vcl/salbtype.hxx>
diff -ur oo_1.1_src.orig/freetype/makefile.mk oo_1.1_src/freetype/makefile.mk
--- oo_1.1_src.orig/freetype/makefile.mk	2003-05-28 07:41:19.000000000 -0500
+++ oo_1.1_src/freetype/makefile.mk	2003-10-24 22:26:01.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=so_freetype
 TARGET=so_freetype
 
+.IF "$(SYSTEM_FREETYPE)" == "YES"
+all:
+	@echo "Using System Installed freetype..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/jpeg/makefile.mk oo_1.1_src/jpeg/makefile.mk
--- oo_1.1_src.orig/jpeg/makefile.mk	2003-03-26 06:25:38.000000000 -0600
+++ oo_1.1_src/jpeg/makefile.mk	2003-10-24 22:30:41.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=jpeg
 TARGET=jpeg
 
+.IF "$(SYSTEM_JPEG)" == "YES"
+all:
+    @echo "Using System Installed jpeg..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/nas/makefile.mk oo_1.1_src/nas/makefile.mk
--- oo_1.1_src.orig/nas/makefile.mk	2003-06-12 04:45:13.000000000 -0500
+++ oo_1.1_src/nas/makefile.mk	2003-10-25 00:05:49.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=nas
 TARGET=nas
 
+.IF "$(SYSTEM_NAS)" == "YES"
+all:
+	@echo "Using System Installed nas..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/neon/neon.patch oo_1.1_src/neon/neon.patch
--- oo_1.1_src.orig/neon/neon.patch	2002-10-24 06:50:58.000000000 -0500
+++ oo_1.1_src/neon/neon.patch	2003-10-25 02:30:32.000000000 -0500
@@ -100,10 +100,10 @@
 ! 
 ! /* Define if you have expat */
 ! #define HAVE_EXPAT 1
-! #define HAVE_OLD_EXPAT 1
+! /* #define HAVE_OLD_EXPAT 1 */
 ! 
 ! /* Define to 1 if you have the <xmlparse.h> header file. */
-! #define HAVE_XMLPARSE_H 1
+! /* #define HAVE_XMLPARSE_H 1 */
 ! 
 ! /* Define if you have libxml */
 ! /* #define HAVE_LIBXML */
@@ -139,7 +139,7 @@
 ***************
 *** 1 ****
 ! dummy
---- 1,49 ----
+--- 1,55 ----
 ! PRJ=..$/..$/..$/..$/..
 ! 
 ! PRJNAME=external
@@ -158,6 +158,12 @@
 ! 
 ! CDEFS+=-DHAVE_CONFIG_H
 ! 
+! .IF "$(SYSTEM_EXPAT)"=="YES"
+! CDEFS+=-DHAVE_EXPAT_H
+! .ELSE
+! CDEFS+=-DHAVE_OLD_EXPAT -DHAVE_XMLPARSE_H
+! .ENDIF
+! 
 ! SLOFILES= \
 !         $(SLO)$/ne_207.obj \
 !         $(SLO)$/ne_acl.obj \
diff -ur oo_1.1_src.orig/package/source/zipapi/Adler32.cxx oo_1.1_src/package/source/zipapi/Adler32.cxx
--- oo_1.1_src.orig/package/source/zipapi/Adler32.cxx	2001-11-15 14:07:35.000000000 -0600
+++ oo_1.1_src/package/source/zipapi/Adler32.cxx	2003-10-25 12:06:12.000000000 -0500
@@ -61,9 +61,13 @@
 #ifndef _ADLER32_HXX
 #include <Adler32.hxx>
 #endif
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#else
 #ifndef _ZLIB_H
 #include <external/zlib/zlib.h>
 #endif
+#endif
 
 /**
  * A class that can be used to compute the Adler-32 checksum of a data
diff -ur oo_1.1_src.orig/package/source/zipapi/CRC32.cxx oo_1.1_src/package/source/zipapi/CRC32.cxx
--- oo_1.1_src.orig/package/source/zipapi/CRC32.cxx	2001-12-04 11:48:19.000000000 -0600
+++ oo_1.1_src/package/source/zipapi/CRC32.cxx	2003-10-25 12:10:15.000000000 -0500
@@ -61,9 +61,13 @@
 #ifndef _CRC32_HXX
 #include <CRC32.hxx>
 #endif
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#else
 #ifndef _ZLIB_H
 #include <external/zlib/zlib.h>
 #endif
+#endif
 #ifndef _PACKAGE_CONSTANTS_HXX_
 #include <PackageConstants.hxx>
 #endif
diff -ur oo_1.1_src.orig/package/source/zipapi/Deflater.cxx oo_1.1_src/package/source/zipapi/Deflater.cxx
--- oo_1.1_src.orig/package/source/zipapi/Deflater.cxx	2001-11-15 14:16:11.000000000 -0600
+++ oo_1.1_src/package/source/zipapi/Deflater.cxx	2003-10-25 12:15:54.000000000 -0500
@@ -61,9 +61,18 @@
 #ifndef _DEFLATER_HXX_
 #include <Deflater.hxx>
 #endif
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#define z_deflateParams deflateParams
+#define z_deflate deflate
+#define z_deflateSetDictionary deflateSetDictionary
+#define z_deflateReset deflateReset
+#define z_deflateEnd deflateEnd
+#else
 #ifndef _ZLIB_H
 #include <external/zlib/zlib.h>
 #endif
+#endif
 #ifndef _VOS_DIAGNOSE_H_
 #include <vos/diagnose.hxx>
 #endif
diff -ur oo_1.1_src.orig/package/source/zipapi/Inflater.cxx oo_1.1_src/package/source/zipapi/Inflater.cxx
--- oo_1.1_src.orig/package/source/zipapi/Inflater.cxx	2001-11-15 14:17:02.000000000 -0600
+++ oo_1.1_src/package/source/zipapi/Inflater.cxx	2003-10-25 12:17:55.000000000 -0500
@@ -61,9 +61,18 @@
 #ifndef _INFLATER_HXX_
 #include <Inflater.hxx>
 #endif
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#define z_inflateSetDictionary inflateSetDictionary
+#define z_inflateSetDictionary inflateSetDictionary
+#define z_inflateReset inflateReset
+#define z_inflateEnd inflateEnd
+#define z_inflate inflate
+#else
 #ifndef _ZLIB_H
 #include <external/zlib/zlib.h>
 #endif
+#endif
 #ifndef _VOS_DIAGNOSE_H_
 #include <vos/diagnose.hxx>
 #endif
diff -ur oo_1.1_src.orig/package/source/zipapi/makefile.mk oo_1.1_src/package/source/zipapi/makefile.mk
--- oo_1.1_src.orig/package/source/zipapi/makefile.mk	2002-01-29 09:30:48.000000000 -0600
+++ oo_1.1_src/package/source/zipapi/makefile.mk	2003-10-25 12:04:45.000000000 -0500
@@ -73,6 +73,10 @@
 .INCLUDE : settings.mk
 .INCLUDE : sv.mk
 
+.IF "$(SYSTEM_ZLIB)" == "YES"
+CFLAGS += -DSYSTEM_ZLIB
+.ENDIF
+
 # --- Files --------------------------------------------------------
 #CFLAGS+=/Ob0 /Od
 SLOFILES= \
diff -ur oo_1.1_src.orig/python/makefile.mk oo_1.1_src/python/makefile.mk
--- oo_1.1_src.orig/python/makefile.mk	2003-08-15 06:22:45.000000000 -0500
+++ oo_1.1_src/python/makefile.mk	2003-10-25 00:27:22.000000000 -0500
@@ -63,6 +63,11 @@
 PRJNAME=so_python
 TARGET=so_python
 
+.IF "$(SYSTEM_PYTHON)" == "YES"
+all:
+	@echo "Using System Installed python..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
diff -ur oo_1.1_src.orig/pyuno/source/loader/makefile.mk oo_1.1_src/pyuno/source/loader/makefile.mk
--- oo_1.1_src.orig/pyuno/source/loader/makefile.mk	2003-08-15 07:38:43.000000000 -0500
+++ oo_1.1_src/pyuno/source/loader/makefile.mk	2003-10-25 13:32:42.000000000 -0500
@@ -69,7 +69,12 @@
 
 DLLPRE = 
 
+.IF "$(SYSTEM_PYTHON)" == "YES"
+CFLAGS += $(SYSTEM_PYTHON_INCLUDES)
+.ELSE
 .INCLUDE :  pyversion.mk
+.ENDIF
+
 #-------------------------------------------------------------------
 
 CFLAGS+=-I$(SOLARINCDIR)$/python
diff -ur oo_1.1_src.orig/pyuno/source/module/makefile.mk oo_1.1_src/pyuno/source/module/makefile.mk
--- oo_1.1_src.orig/pyuno/source/module/makefile.mk	2003-08-15 07:39:44.000000000 -0500
+++ oo_1.1_src/pyuno/source/module/makefile.mk	2003-10-25 13:25:39.000000000 -0500
@@ -66,7 +66,11 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
+.IF "$(SYSTEM_PYTHON)" == "YES"
+CFLAGS += $(SYSTEM_PYTHON_INCLUDES)
+.ELSE
 .INCLUDE :  pyversion.mk
+.ENDIF
 #-------------------------------------------------------------------
 
 CFLAGS+=-I$(SOLARINCDIR)$/python
diff -ur oo_1.1_src.orig/pyuno/zipcore/makefile.mk oo_1.1_src/pyuno/zipcore/makefile.mk
--- oo_1.1_src.orig/pyuno/zipcore/makefile.mk	2003-08-15 07:45:45.000000000 -0500
+++ oo_1.1_src/pyuno/zipcore/makefile.mk	2003-10-25 13:09:27.000000000 -0500
@@ -1,6 +1,10 @@
 PRJNAME=pyuno
 PRJ=..
 
+.IF "$(SYSTEM_PYTHON)" == "YES"
+all:
+	@echo "Using system installed python..."
+.ELSE
 .INCLUDE : settings.mk
 .INCLUDE : pyversion.mk
 
@@ -48,3 +52,4 @@
 .IF "$(GUI)"== "UNX"
 	chmod +x $@
 .ENDIF
+.ENDIF
diff -ur oo_1.1_src.orig/sablot/Sablot-0.52.patch oo_1.1_src/sablot/Sablot-0.52.patch
--- oo_1.1_src.orig/sablot/Sablot-0.52.patch	2003-08-05 04:41:11.000000000 -0500
+++ oo_1.1_src/sablot/Sablot-0.52.patch	2003-10-25 01:33:48.000000000 -0500
@@ -3,7 +3,7 @@
 ***************
 *** 1 ****
 ! dummy
---- 1,163 ----
+--- 1,167 ----
 ! #*************************************************************************
 ! #
 ! #   $RCSfile: openoffice-1.1.0-use-system-components-001.patch,v $
@@ -77,6 +77,10 @@
 ! 
 ! .INCLUDE: settings.mk
 ! 
+! .IF "$(SYSTEM_EXPAT)"=="YES"
+! CFLAGS+=-DHAVE_EXPAT_H
+! .ENDIF
+! 
 ! .IF "$(GUI)"=="WNT"
 ! CFLAGS+=-GR
 ! .ENDIF
diff -ur oo_1.1_src.orig/sax/source/expatwrap/makefile.mk oo_1.1_src/sax/source/expatwrap/makefile.mk
--- oo_1.1_src.orig/sax/source/expatwrap/makefile.mk	2001-03-02 01:16:52.000000000 -0600
+++ oo_1.1_src/sax/source/expatwrap/makefile.mk	2003-10-25 10:07:53.000000000 -0500
@@ -72,6 +72,10 @@
 #-----------------------------------------------------------
 CFLAGS += -DXML_UNICODE
 
+.IF "$(SYSTEM_EXPAT)" == "YES"
+CFLAGS += -DSYSTEM_EXPAT
+.ENDIF
+
 SLOFILES =\
 		$(SLO)$/xml2utf.obj\
 		$(SLO)$/attrlistimpl.obj\
diff -ur oo_1.1_src.orig/sax/source/expatwrap/sax_expat.cxx oo_1.1_src/sax/source/expatwrap/sax_expat.cxx
--- oo_1.1_src.orig/sax/source/expatwrap/sax_expat.cxx	2002-05-15 11:33:29.000000000 -0500
+++ oo_1.1_src/sax/source/expatwrap/sax_expat.cxx	2003-10-25 10:04:11.000000000 -0500
@@ -82,7 +82,11 @@
 #include <cppuhelper/implbase1.hxx>
 #include <cppuhelper/implbase2.hxx>
 
+#ifdef SYSTEM_EXPAT
+#include <expat.h>
+#else
 #include "expat/xmlparse.h"
+#endif
 
 using namespace ::rtl;
 using namespace ::std;
diff -ur oo_1.1_src.orig/scp/source/global/makefile.mk oo_1.1_src/scp/source/global/makefile.mk
--- oo_1.1_src.orig/scp/source/global/makefile.mk	2003-04-17 10:42:01.000000000 -0500
+++ oo_1.1_src/scp/source/global/makefile.mk	2003-10-26 15:25:00.000000000 -0600
@@ -79,6 +79,10 @@
 SCPDEFS+=-DINCLUDE_JAVA_ACCESSBRIDGE
 .ENDIF
 
+.IF "$(SYSTEM_FREETYPE)" == "YES"
+SCPDEFS += -DSYSTEM_FREETYPE
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl fat fam ada adm web wem srv ofs omc
 PARFILES= \
 		setupzip.par		\
diff -ur oo_1.1_src.orig/scp/source/global/setupzip.scp oo_1.1_src/scp/source/global/setupzip.scp
--- oo_1.1_src.orig/scp/source/global/setupzip.scp	2003-08-22 06:44:24.000000000 -0500
+++ oo_1.1_src/scp/source/global/setupzip.scp	2003-10-26 14:24:46.000000000 -0600
@@ -559,6 +559,7 @@
 #ifdef UNX
 
 #ifndef MACOSX
+#ifndef SYSTEM_FREETYPE
 File GID_FILE_LIB_FREETYPE
     #if defined (LINUX)  &&  ( defined (CLASSIC_FAT_PRODUCT) || defined (OSL_PRODUCT) || defined (FAM_PRODUCT) )
         Dir     = GID_DIR_FILTER;
@@ -574,6 +575,7 @@
     #endif
 End
 #endif
+#endif
 
 #ifdef OSL_PRODUCT
 STD_FONTUNX_FILE( GID_FILE_FNT_VERA, Vera.ttf, Vera Sans)
@@ -654,4 +656,4 @@
         Name (99)       = "fontunxpsprint.zip";
 End
 
-#endif
\ No newline at end of file
+#endif
diff -ur oo_1.1_src.orig/scp/source/global/udk_base.scp oo_1.1_src/scp/source/global/udk_base.scp
--- oo_1.1_src.orig/scp/source/global/udk_base.scp	2003-07-16 13:15:15.000000000 -0500
+++ oo_1.1_src/scp/source/global/udk_base.scp	2003-10-26 12:20:02.000000000 -0600
@@ -6,7 +6,7 @@
 // Temporary solution/hack: at the moment libstdc++.so and libgcc_s.so are
 // needed for unxlngi4 environment (setup is linked against it).
 
-#if ! (defined (FREEBSD) || defined(MACOSX) )
+#if ! (defined (FREEBSD) || defined(MACOSX) || defined(LINUX) )
 #if ( defined( _gcc3 ) && ( defined( C300 ) || defined( C322 ) ))
 
 File gid_File_Lib_gcc
diff -ur oo_1.1_src.orig/scp/source/office/files.scp oo_1.1_src/scp/source/office/files.scp
--- oo_1.1_src.orig/scp/source/office/files.scp	2003-10-02 21:50:58.000000000 -0500
+++ oo_1.1_src/scp/source/office/files.scp	2003-10-26 20:58:34.000000000 -0600
@@ -876,6 +876,7 @@
         #endif
 End
 
+#ifndef SYSTEM_CURL
 File GID_FILE_LIB_CURL
         TXT_FILE_BODY;
         Styles          = (PACKED);
@@ -886,6 +887,7 @@
         Name            = "libcurl.dll";
         #endif
 End
+#endif
 
 
 File GID_FILE_LIB_I18NREGEXP
@@ -933,6 +935,8 @@
 
 #endif
 
+#ifndef SYSTEM_DB
+
 File GID_FILE_LIB_DB31
         TXT_FILE_BODY;
         Styles          = (PACKED);
@@ -970,6 +974,8 @@
 
 #endif
 
+#endif
+
 #ifdef UNX
 
 File GID_FILE_MSGBOX_GNOME
@@ -3176,14 +3182,6 @@
 
 #endif
                                   // of #ifdef SOLAR_JAVA
-#ifndef MACOSX
-File GID_FILE_MOZILLA_RUNTIME
-        TXT_FILE_BODY;
-        Styles = (ARCHIVE);
-        Dir    = GID_DIR_PROGRAM;
-        Name	= "mozruntime.zip";
-End
-#endif
 
 File GID_FILE_HTML_THIRDPARTYLICENSEREADME
         Name   = "THIRDPARTYLICENSEREADME.html";
diff -ur oo_1.1_src.orig/scp/source/office/makefile.mk oo_1.1_src/scp/source/office/makefile.mk
--- oo_1.1_src.orig/scp/source/office/makefile.mk	2003-07-16 13:17:16.000000000 -0500
+++ oo_1.1_src/scp/source/office/makefile.mk	2003-10-26 20:57:13.000000000 -0600
@@ -72,6 +72,14 @@
 SCPDEFS+=-DENABLE_CRASHDUMP
 .ENDIF
 
+.IF "$(SYSTEM_DB)" == "YES"
+SCPDEFS += -DSYSTEM_DB
+.ENDIF
+
+.IF "$(SYSTEM_CURL)" == "YES"
+SCPDEFS += -DSYSTEM_CURL
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  svpre.mk
diff -ur oo_1.1_src.orig/scp/source/python/makefile.mk oo_1.1_src/scp/source/python/makefile.mk
--- oo_1.1_src.orig/scp/source/python/makefile.mk	2003-07-02 10:15:41.000000000 -0500
+++ oo_1.1_src/scp/source/python/makefile.mk	2003-10-26 21:27:56.000000000 -0600
@@ -76,6 +76,10 @@
 
 SCP_PRODUCT_TYPE=osl
 
+.IF "$(SYSTEM_PYTHON)" == "YES"
+SCPDEFS += -DSYSTEM_PYTHON
+.ENDIF
+
 .IF "$(GUI)"=="UNX"
 SCRIPTFILES=	unxbasic_python.par
 .ELSE
diff -ur oo_1.1_src.orig/scp/source/python/python.scp oo_1.1_src/scp/source/python/python.scp
--- oo_1.1_src.orig/scp/source/python/python.scp	2003-08-18 09:32:57.000000000 -0500
+++ oo_1.1_src/scp/source/python/python.scp	2003-10-26 21:53:35.000000000 -0600
@@ -160,6 +160,7 @@
     Value  = "$ORIGIN/services.rdb";
 End
 
+#ifdef SYSTEM_PYTHON
 File gid_File_Py_python_core
      TXT_FILE_BODY;
      ModuleID = GID_MODULE_PYUNO;
@@ -167,6 +168,7 @@
      Name = "python-core-2.2.2.zip";
      Styles          = (ARCHIVE);
 End
+#endif
 
 Profile gid_Profile_pythonloader_uno_ini
     ModuleID = GID_MODULE_PYUNO;
@@ -214,6 +216,7 @@
 
 #ifdef UNX
 #ifndef MACOSX
+#ifndef SYSTEM_PYTHON
 File gid_File_Lib_Python_so
      TXT_FILE_BODY;
      ModuleID = GID_MODULE_PYUNO;
@@ -229,6 +232,7 @@
     Name = STRING(CONCAT3(libpython,UNXSUFFIX,.2));
     Styles = (NETWORK,RELATIVE);
 End
+#endif
 #else
 File gid_File_Lib_Python_so
      TXT_FILE_BODY;
@@ -247,6 +251,7 @@
 End
 #endif
 
+#ifndef SYSTEM_PYTHON
 Shortcut gid_Shortcut_Lib_Python_2
     ModuleID =GID_MODULE_PYUNO;
     FileID = gid_File_Lib_Python_so;
@@ -254,6 +259,7 @@
     Name = STRING(CONCAT2(libpython,UNXSUFFIX));
     Styles = (NETWORK,RELATIVE);
 End
+#endif
 
 
 File gid_File_Pyuno
@@ -264,6 +270,7 @@
     Styles          = (PACKED);
 End
 
+#ifndef SYSTEM_PYTHON
 File gid_File_python_sh
      BIN_FILE_BODY;
      ModuleID = GID_MODULE_PYUNO;
@@ -271,5 +278,6 @@
      Dir = GCFG_BINARY_DIR;
      Styles = (PACKED);
 End
+#endif
 
 #endif
diff -ur oo_1.1_src.orig/solenv/inc/libs.mk oo_1.1_src/solenv/inc/libs.mk
--- oo_1.1_src.orig/solenv/inc/libs.mk	2003-07-02 08:43:33.000000000 -0500
+++ oo_1.1_src/solenv/inc/libs.mk	2003-10-25 22:58:53.000000000 -0500
@@ -171,13 +171,31 @@
 HM2LIBST=-lhmwrap
 LINGULIB=$(HM2LIBST)
 LNGLIB=-llng$(OFFICEUPD)$(DLLPOSTFIX)
+.IF "$(SYSTEM_EXPAT)" == "YES"
+EXPAT3RDLIB=-lexpat
+EXPATASCII3RDLIB=-lexpat
+.ELSE
 EXPAT3RDLIB=-lexpat_xmlparse -lexpat_xmltok
 EXPATASCII3RDLIB=-lascii_expat_xmlparse -lexpat_xmltok
+.ENDIF
+.IF "$(SYSTEM_ZLIB)" == "YES"
+ZLIB3RDLIB=-lz
+.ELSE
 ZLIB3RDLIB=-lzlib
+.ENDIF
+.IF "$(SYSTEM_JPEG)" == "YES"
+JPEG3RDLIB=-ljpeg
+.ELSE
 JPEG3RDLIB=-ljpeglib
+.ENDIF
 NEON3RDLIB=-lneon
+.IF "$(SYSTEM_DB)" == "YES"
+BERKELEYLIB=-ldb-$(SYSTEM_DB_VERSION)
+BERKELEYCPPLIB=-ldb_cxx-$(SYSTEM_DB_VERSION)
+.ELSE
 BERKELEYLIB=-ldb-3.2
 BERKELEYCPPLIB=-ldb_cxx-3.2
+.ENDIF
 CURLLIB=-lcurl
 SFX2LIB=-lsfx$(OFFICEUPD)$(DLLPOSTFIX)
 SFXLIB=-lsfx$(OFFICEUPD)$(DLLPOSTFIX)
diff -ur oo_1.1_src.orig/tools/source/zcodec/makefile.mk oo_1.1_src/tools/source/zcodec/makefile.mk
--- oo_1.1_src.orig/tools/source/zcodec/makefile.mk	2000-09-18 12:03:10.000000000 -0500
+++ oo_1.1_src/tools/source/zcodec/makefile.mk	2003-10-25 11:05:13.000000000 -0500
@@ -71,6 +71,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE :  sv.mk
 
+.IF "$(SYSTEM_ZLIB)" == "YES"
+CFLAGS += -DSYSTEM_ZLIB
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=	$(SLO)$/zcodec.obj
diff -ur oo_1.1_src.orig/tools/source/zcodec/zcodec.cxx oo_1.1_src/tools/source/zcodec/zcodec.cxx
--- oo_1.1_src.orig/tools/source/zcodec/zcodec.cxx	2000-09-18 12:03:10.000000000 -0500
+++ oo_1.1_src/tools/source/zcodec/zcodec.cxx	2003-10-25 11:06:32.000000000 -0500
@@ -62,9 +62,13 @@
 #ifndef _STREAM_HXX
 #include "stream.hxx"
 #endif
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#else
 #ifndef _ZLIB_H
 #include "zlib/zlib.h"
 #endif 
+#endif
 #ifndef _ZCODEC_HXX
 #include "zcodec.hxx"
 #endif
diff -ur oo_1.1_src.orig/vcl/prj/build.lst oo_1.1_src/vcl/prj/build.lst
--- oo_1.1_src.orig/vcl/prj/build.lst	2003-06-12 05:24:40.000000000 -0500
+++ oo_1.1_src/vcl/prj/build.lst	2003-10-25 00:33:37.000000000 -0500
@@ -1,4 +1,4 @@
-vc	vcl	:	nas freetype psprint rsc sot ucbhelper unotools rvpapi icu unoil ridljar x11_extensions NULL
+vc	vcl	:	nas freetype psprint rsc sot ucbhelper unotools rvpapi icu unoil ridljar NULL
 vc	vcl										usr1	-	all	vc_mkout NULL
 vc	vcl\source\unotypes						nmake	-	all	vc_unot NULL
 vc	vcl\source\glyphs						nmake	-	all	vc_glyphs vc_unot NULL
diff -ur oo_1.1_src.orig/vcl/source/glyphs/makefile.mk oo_1.1_src/vcl/source/glyphs/makefile.mk
--- oo_1.1_src.orig/vcl/source/glyphs/makefile.mk	2002-02-15 09:52:27.000000000 -0600
+++ oo_1.1_src/vcl/source/glyphs/makefile.mk	2003-10-25 19:15:27.000000000 -0500
@@ -70,6 +70,10 @@
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 .INCLUDE :  settings.mk
 
+.IF "$(SYSTEM_FREETYPE)" == "YES"
+CFLAGS += $(SYSTEM_FREETYPE_INCLUDES)
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 .IF "$(USE_BUILTIN_RASTERIZER)" != ""
diff -ur oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/CreateDb.java oo_1.1_src/xmlhelp/source/com/sun/star/help/CreateDb.java
--- oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/CreateDb.java	2000-11-20 06:08:42.000000000 -0600
+++ oo_1.1_src/xmlhelp/source/com/sun/star/help/CreateDb.java	2003-10-26 08:33:44.000000000 -0600
@@ -37,7 +37,7 @@
 		table.set_errpfx( "HelpAccess" );
 		try
 		{
-			table.open( dbName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
+			table.open( null,dbName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
 			Dbc cursor = table.cursor( null,0 );
 
 			StringDbt key = new StringDbt();
@@ -131,7 +131,7 @@
       		Db table = new Db(null, 0);
       		table.set_error_stream(System.err);
       		table.set_errpfx("AccessExample");
-      		table.open("e:/rvp603b/help/helpaccess.db", null, Db.DB_HASH, Db.DB_CREATE, 0644);
+      		table.open(null,"e:/rvp603b/help/helpaccess.db", null, Db.DB_HASH, Db.DB_CREATE, 0644);
       		StringDbt key = new StringDbt(key1);
       		StringDbt data = new StringDbt(data1);
       		try
diff -ur oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/HelpDatabases.java oo_1.1_src/xmlhelp/source/com/sun/star/help/HelpDatabases.java
--- oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/HelpDatabases.java	2003-03-27 12:07:28.000000000 -0600
+++ oo_1.1_src/xmlhelp/source/com/sun/star/help/HelpDatabases.java	2003-10-26 08:33:44.000000000 -0600
@@ -239,7 +239,7 @@
 				table = new Db( null,0 );
 
 				String tablePath = _installDirectory + key + ".db";
-				table.open( tablePath,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
+				table.open( null,tablePath,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
 				_dbHash.put( key,table );
             }
             catch( DbException e )
@@ -274,7 +274,7 @@
 				table = new Db( null,0 );
 
 				String tablePath = _installDirectory + key + ".ht";
-				table.open( tablePath,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
+				table.open( null,tablePath,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
 				_dbHash.put( key,table );
             }
             catch( DbException e )
@@ -517,7 +517,7 @@
 				String fileName = HelpDatabases.getInstallDirectory() + keyStr + ".key";
         		Db table = new Db( null,0 );
 				System.err.println( fileName );
-				table.open( fileName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
+				table.open( null,fileName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
 				Dbc cursor = table.cursor( null,0 );
 				StringDbt key = new StringDbt();
 				StringDbt data = new StringDbt();
diff -ur oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/HelpIndexer.java oo_1.1_src/xmlhelp/source/com/sun/star/help/HelpIndexer.java
--- oo_1.1_src.orig/xmlhelp/source/com/sun/star/help/HelpIndexer.java	2003-03-27 12:07:28.000000000 -0600
+++ oo_1.1_src/xmlhelp/source/com/sun/star/help/HelpIndexer.java	2003-10-26 08:33:44.000000000 -0600
@@ -150,7 +150,7 @@
                     		+ _module
                        		+ ".db";
 
-			table.open( fileName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
+			table.open( null,fileName,null,Db.DB_BTREE,Db.DB_RDONLY,0644 );
 			Dbc cursor = table.cursor( null,0 );
 			StringDbt key = new StringDbt();
 			StringDbt data = new StringDbt();
@@ -440,7 +440,7 @@
                     		      + _module
                        		      + ".key";
 
-				table.open( fileName,null,Db.DB_BTREE,Db.DB_CREATE,0644 );
+				table.open( null,fileName,null,Db.DB_BTREE,Db.DB_CREATE,0644 );
 
             	for( int i = 0; i < list.length; ++i )
             	{
@@ -481,7 +481,7 @@
                     		+ _module
                        		+ ".ht";
 
-			table.open( fileName,null,Db.DB_BTREE,Db.DB_CREATE,0644 );
+			table.open( null,fileName,null,Db.DB_BTREE,Db.DB_CREATE,0644 );
 
             for( int i = 0; i < list.length; ++i )
             {
diff -ur oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/databases.cxx oo_1.1_src/xmlhelp/source/cxxhelp/provider/databases.cxx
--- oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/databases.cxx	2003-04-04 11:09:36.000000000 -0600
+++ oo_1.1_src/xmlhelp/source/cxxhelp/provider/databases.cxx	2003-10-25 20:27:52.000000000 -0500
@@ -60,7 +60,12 @@
  ************************************************************************/
 
 
+#ifdef SYSTEM_DB
+#include <db_cxx.h>
+#define DB_VER (1000*DB_VERSION_MAJOR + 100*DB_VERSION_MINOR)
+#else
 #include <berkeleydb/db_cxx.h>
+#endif
 #ifndef _VOS_DIAGNOSE_HXX_
 #include <vos/diagnose.hxx>
 #endif
@@ -497,7 +502,15 @@
 		
 		rtl::OString fileName( fileNameOU.getStr(),fileNameOU.getLength(),osl_getThreadTextEncoding() );
 		
+#ifdef SYSTEM_DB
+#if (DB_VER >= 4100)
+		if( table->open( NULL,fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#else
+		if( table->open( fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#endif
+#else
 		if( table->open( fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#endif
 		{
 			table->close( 0 );
 			delete table;
@@ -711,7 +724,15 @@
                                osl_getThreadTextEncoding() );
 		
 		Db table(0,DB_CXX_NO_EXCEPTIONS);
+#ifdef SYSTEM_DB
+#if (DB_VER >= 4100)
+		if( 0 == table.open( NULL,fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#else
 		if( 0 == table.open( fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#endif
+#else
+		if( 0 == table.open( fileName.getStr(),0,DB_BTREE,DB_RDONLY,0644 ) )
+#endif
 		{   
 			std::vector<KeywordInfo::KeywordElement> aVector;
 			Db* idmap = getBerkeley( Database,Language );
diff -ur oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/makefile.mk oo_1.1_src/xmlhelp/source/cxxhelp/provider/makefile.mk
--- oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/makefile.mk	2003-04-28 11:19:35.000000000 -0500
+++ oo_1.1_src/xmlhelp/source/cxxhelp/provider/makefile.mk	2003-10-25 20:29:18.000000000 -0500
@@ -75,6 +75,10 @@
 
 CFLAGS +=  -DHAVE_EXPAT_H
 
+.IF "$(SYSTEM_DB)" == "YES"
+CFLAGS += -DSYSTEM_DB
+.ENDIF
+
 .IF "$(GUI)"=="WNT"
 CFLAGS+=-GR
 .ENDIF
diff -ur oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/urlparameter.cxx oo_1.1_src/xmlhelp/source/cxxhelp/provider/urlparameter.cxx
--- oo_1.1_src.orig/xmlhelp/source/cxxhelp/provider/urlparameter.cxx	2003-08-05 04:42:33.000000000 -0500
+++ oo_1.1_src/xmlhelp/source/cxxhelp/provider/urlparameter.cxx	2003-10-25 16:34:38.000000000 -0500
@@ -93,9 +93,13 @@
 #ifndef ShandlerHIncl
 #include <sablot/shandler.h>
 #endif
+#ifdef SYSTEM_DB
+#include <db_cxx.h>
+#else
 #ifndef _DB_CXX_H_
 #include <berkeleydb/db_cxx.h>
 #endif
+#endif
 #ifndef _URLPARAMETER_HXX_
 #include <provider/urlparameter.hxx>
 #endif
diff -ur oo_1.1_src.orig/xmlhelp/source/treeview/makefile.mk oo_1.1_src/xmlhelp/source/treeview/makefile.mk
--- oo_1.1_src.orig/xmlhelp/source/treeview/makefile.mk	2001-07-18 05:30:44.000000000 -0500
+++ oo_1.1_src/xmlhelp/source/treeview/makefile.mk	2003-10-25 14:20:02.000000000 -0500
@@ -72,6 +72,10 @@
 
 .INCLUDE: settings.mk
 
+.IF "$(SYSTEM_EXPAT)" == "YES"
+CFLAGS += -DSYSTEM_EXPAT
+.ENDIF
+
 SLOFILES=\
 	$(SLO)$/tvfactory.obj  \
 	$(SLO)$/tvread.obj
diff -ur oo_1.1_src.orig/xmlhelp/source/treeview/tvread.cxx oo_1.1_src/xmlhelp/source/treeview/tvread.cxx
--- oo_1.1_src.orig/xmlhelp/source/treeview/tvread.cxx	2001-11-08 10:42:46.000000000 -0600
+++ oo_1.1_src/xmlhelp/source/treeview/tvread.cxx	2003-10-25 14:19:06.000000000 -0500
@@ -8,9 +8,13 @@
 #ifndef _TREEVIEW_TVREAD_HXX_
 #include "tvread.hxx"
 #endif
+#ifdef SYSTEM_EXPAT
+#include <expat.h>
+#else
 #ifndef XmlParse_INCLUDED
 #include <expat/xmlparse.h>
 #endif
+#endif
 #ifndef _OSL_FILE_HXX_
 #include <osl/file.hxx>
 #endif
diff -ur oo_1.1_src.orig/zlib/makefile.mk oo_1.1_src/zlib/makefile.mk
--- oo_1.1_src.orig/zlib/makefile.mk	2003-03-27 12:13:15.000000000 -0600
+++ oo_1.1_src/zlib/makefile.mk	2003-10-25 00:34:35.000000000 -0500
@@ -65,6 +65,11 @@
 PRJNAME=zlib
 TARGET=zlib
 
+.IF "$(SYSTEM_ZLIB)" == "YES"
+all:
+	@echo "Using System Installed zlib..."
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
